metaserver/src/main/scala/scala/meta/languageserver/Compiler.scala

```diff
@@ -47,41 +54,97 @@ class Compiler(
         )
       }
 
+  def signatureHelp(
+      path: AbsolutePath,
+      line: Int,
+      column: Int
+  ): SignatureHelp = {
+    getCompiler(path, line, column - 1).fold(noSignatureHelp) {
+      case (compiler, position) =>
+        logger.info(s"Signature help at $path:$line:$column")
+        // Find the last leading open paren to get the method symbol
+        // of this argument list. Note, this may still cause false positives
+        // in cases like `foo(bar(), <cursor>)` since we will find the
+        // symbol of `bar` when we want to match against `foo`.
+        // Related https://github.com/scalameta/language-server/issues/52
+        val lastParenOffset =
+          position.source.content.lastIndexOf('(', position.point)
```

Comment:
This works outside of parentheses too 😄 

<img width="300" alt="screen shot 2017-11-23 at 16 39 20" src="https://user-images.githubusercontent.com/766656/33180825-590a60c8-d06e-11e7-8525-557fd1b0b400.png">